cpp/include/cudf/partitioning.hpp

```diff
@@ -17,6 +17,7 @@
 #pragma once
 
 #include <cudf/types.hpp>
+#include <cudf/table/table.hpp>
```

Comment:
If the benchmark needs this include, then the benchmark should include it, not `partitioning.hpp`.

Reply:
My thinking was that `partitioning.hpp` uses types like `std::unique_ptr<table>`, which require a full declaration of `table`, not just the forward declaration from `types.hpp`, so `partitioning.hpp` would need to include `table.hpp` in order to be self sufficient. Otherwise you'd need to make sure to include `table.hpp` before `partitioning.hpp` in each TU where or appears, right?

Reply:
The goal is to keep includes top-level headers as minimal as possible to reduce compile time and avoid "leaking" third party includes to top-level headers.

Reply:
> Otherwise you'd need to make sure to include table.hpp before partitioning.hpp in each TU where it appears, right?

You'd need to include both of them, yes, but not in any particular order.

Reply:
Ah, my mistake; `partitioning.hpp` does compile with just the forward declaration as you said. Updated the PR